### ISSUE TITLE
Questions about the Latent Watermark Pre-training stage

### ISSUE THREAD

**Issue by LiRunyi2001** [Georgefwt/AquaLoRA#1]
Hi there! I've trained the secret encoder and decoder in the Latent Watermark Pre-training stage using such shell script:
`python latent_wm_pretrain.py --output_dir test --epochs 40 --dataset coco2017/train2017`
and it turned out that the bit accuracy was about 50%, which means that the secret encoder failed to learn.
The output log is below:
![image](https://github.com/Georgefwt/AquaLoRA/assets/54633216/90a244bb-727a-4896-ba4f-2308e9d268b7)
Is there anything where I've done wrong? Thanks a lot!

**Comment by Georgefwt** [comment]
![image](https://github.com/Georgefwt/AquaLoRA/assets/49113320/2cc70cb1-31b8-4f79-b511-c5a85fcbaf04)
I tested it locally, and the loss is decreasing normally. Could you show me your loss curve so I can understand the specific situation? Additionally, you can check out my latest released checkpoint.

**Comment by LiRunyi2001** [comment]
Thanks for your reply! My loss is oscillating around 0.69 with no obvious signs of convergence.
![image](https://github.com/Georgefwt/AquaLoRA/assets/54633216/7f7957d5-5915-4044-9597-c4ab96143661)
The logged lpips loss is about 0.01, msgloss 0.69, and prvl_loss about 0.1 to 0.2. 
Detailed training setting: batch size 4, epochs 40, learning rate 0.001, bit_num 48. By the way I set hyper-parameter fixinit as False, will this effect the training process?

**Comment by Georgefwt** [comment]
Yes, when starting training from scratch, fixinit should be set to True (this is also the default in the code). Only when resuming training should fixinit be set to False.

**Comment by LiRunyi2001** [comment]
Oh thanks! And I am curious about the function of fixinit, as it will set the input image as zero. Why is this setting needed, and the decoded image is also weird.
The initial saved decoed image is a gray image:
![image](https://github.com/Georgefwt/AquaLoRA/assets/54633216/320cc1e4-f0bc-4dcb-bb7d-80f4cbb98840)
and after a few epochs it turned into:
![image](https://github.com/Georgefwt/AquaLoRA/assets/54633216/91ff406a-7ebb-40f2-901c-8493bb97bded)

**Comment by Georgefwt** [comment]
This is expected, as there is only message loss at the beginning of training. Just train for more epochs, and the output will be normal.

**Comment by LiRunyi2001** [comment]
Ok thanks a lot!
